website/source/api/secret/ad/index.html.md

```diff
@@ -0,0 +1,174 @@
+* `starttls` (bool, optional) - If true, issues a `StartTLS` command after establishing an unencrypted connection.
+* `insecure_tls` - (bool, optional) - If true, skips LDAP server SSL certificate verification - insecure, use with caution!
+* `certificate` - (string, optional) - CA certificate to use when verifying LDAP server certificate, must be x509 PEM encoded.
+
+### Binding parameters
+
+* `binddn` (string, required) - Distinguished name of object to bind when performing user and group search. Example: `cn=vault,ou=Users,dc=example,dc=com`
+* `bindpass` (string, required) - Password to use along with `binddn` when performing user search.
+* `userdn` (string, optional) - Base DN under which to perform user search. Example: `ou=Users,dc=example,dc=com`
+* `upndomain` (string, optional) - userPrincipalDomain used to construct the UPN string for the authenticating user. The constructed UPN will appear as `[username]@UPNDomain`. Example: `example.com`, which will cause vault to bind as `username@example.com`.
+
+## Config management
+
+At present, this endpoint does not confirm that the provided AD credentials are
+valid AD credentials with proper permissions.
+
+| Method   | Path                   | Produces               |
+| :------- | :--------------------- | :--------------------- |
+| `POST`   | `/ad/config`           | `204 (empty body)`     |
+| `GET`    | `/ad/config`           | `200 (see body below)` |
+| `DELETE` | `/ad/config`           | `204 (empty body)`     |
+
+### Sample Payload
+
+```json
+{
+  "binddn": "domain-admin",
+  "bindpass": "pa$$w0rd",
+  "url": "ldap://127.0.0.11",
+  "userdn": "dc=example,dc=com"
+}
+```
+
+### Sample Post Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/ad/config
+```
+### Sample Get Response
+
+```json
+{
+  "binddn": "domain-admin",
+  "url": "ldaps://example.com",
+  "certificate": "-----BEGIN CERTIFICATE-----....",
+  "userdn": "dc=example,dc=com",
+  "insecure_tls": false,
+  "start_tls": true,
+  "tls_min_version": "tls12",
+  "tls_max_version": "tls12",
+  "ttl": 2764800,
+  "max_ttl": 2764800,
+  "password_length": 64
+}
+
+```
+
+## Role management
+
+The `roles` endpoint configures how Vault will manage the passwords for individual service accounts.
+
+### Parameters
+
+* `service_account_name` (string, required) - The name of a pre-existing service account in Active Directory that maps to this role.
+* `ttl` (string, optional) - The password time-to-live in seconds. Defaults to the configuration `ttl` if not provided.
+
+When adding a role, Vault verifies its associated service account exists.
+
+| Method   | Path                   | Produces               |
+| :------- | :--------------------- | :--------------------- |
+| `GET`    | `/ad/roles`            | `204 (see body below)` |
+| `POST`   | `/ad/roles/:role_name` | `204 (empty body)`     |
```

Comment:
Can we split these out since the payload and responses are not consistent across all these different paths?

Reply:
I sorta do break out the GET and LIST response bodies below. I'm not sure what you mean - like I think you must be wanting me to format the examples differently but I'm not sure how?